app/views/posts/_post.html.erb

```diff
@@ -0,0 +1,11 @@
+  <div class="media">
+    <div class="media-body">
+      <h4 class="media-heading">
+        <%= link_to post.title, topic_post_path(@topic, post) %>
```

Comment:
You shouldn't use any controller / instance variables here (`@topic`.

The goal is a partial can be embedded anywhere, including a controller action which might need to support multiple topics.

Instead you can either
1. Pass in a local `topic` variable like you did for `post`
2. Take advantage of associates to use `post.topic` (probably the best choice)